### PR TITLE
Improve android-reviewer skill accuracy and update gh-aw

### PR DESCRIPTION
**Pull request by jonathanpeppers**
The AI reviewer on PR #11415 produced two false positives: it suggested using the `IsNullOrEmpty()` extension method in a .NET 11 console app that doesn't reference the assembly defining it, and it summarized suggestions only in the review body instead of posting inline comments.

This PR fixes the root causes and updates the gh-aw tooling.

## Changes

- **Update gh-aw** from v0.68.3 to v0.74.8, run `gh aw upgrade` codemods
- **Scope `IsNullOrEmpty()` rule to netstandard2.0 projects** -- the `NullableExtensions` class exists as a workaround for netstandard2.0's BCL lacking `[NotNullWhen]` on `string.IsNullOrEmpty`. In modern .NET (.NET 10+), the static method is already properly annotated, so the extension is unnecessary. The rule now says to verify `TargetFramework` before flagging.
- **Add "verify project context" constraint** -- reviewer must check `TargetFramework`, references, and available APIs before applying rules. A rule about an extension method is irrelevant if the project doesn't reference the assembly that defines it.
- **Require inline comments, not body-only summaries** -- suggestions must be posted as inline comments on changed lines. If a suggestion can't be attached to a changed line, omit it rather than burying it in the summary.

- [x] Useful description of *why the change is necessary*.
- [ ] Links to issues fixed
- [ ] Unit tests

### CODE REVIEW

File: .github/workflows/nightly-fix-finder.md [hunk 1]
```diff
@@ -1,156 +1,159 @@
 ---
```
Comment (jonathanpeppers) [inline .github/workflows/nightly-fix-finder.md:1]: `gh aw upgrade` (v0.68.3 -> v0.74.8) re-serializes YAML frontmatter into a canonical format: alphabetical keys, block-style arrays, stripped unnecessary quotes. This is intentional normalization from the upgrade codemods, not a manual change.

File: .github/workflows/copilot-setup-steps.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ jobs:
         fetch-depth: 1
 
     - name: Install gh-aw extension
-      uses: github/gh-aw-actions/setup-cli@ba90f2186d7ad780ec640f364005fa24e797b360 # v0.68.3
+      uses: github/gh-aw-actions/setup-cli@efa55847f72aadb03490d955263ff911bf758700 # v0.74.8
       with:
         version: v0.68.3
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply .github/workflows/copilot-setup-steps.yml:33]: The `setup-cli` action version in `actions-lock.json` was updated automatically by `gh aw upgrade`. `copilot-setup-steps.yml` is the bootstrap workflow and its version is managed independently -- these are not expected to match.

File: .github/skills/android-reviewer/SKILL.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ Flag severity clearly in every comment:
...
 
-**Every review should produce at least one inline comment.** Even clean PRs have opportunities for improvement — code consolidation, missing edge-case tests, perf micro-optimizations, or documentation gaps. Use 💡 suggestions for these. A review with zero comments appears superficial and misses the chance to share knowledge. Only omit inline comments if the PR is truly trivial (e.g., a 1-line typo fix or dependency bump).
+**Every review should produce at least one inline comment.** Even clean PRs have opportunities for improvement — code consolidation, missing edge-case tests, perf micro-optimizations, or documentation gaps. Use 💡 suggestions for these. A review with zero inline comments appears superficial and misses the chance to share knowledge. Only omit inline comments if the PR is truly trivial (e.g., a 1-line typo fix or dependency bump). **Do NOT summarize suggestions only in the review body — post them as inline comments on the relevant line.** If a suggestion cannot be posted inline (e.g., it's about pre-existing code or missing code), either find the closest relevant changed line to attach it to, or omit it entirely rather than burying it in the summary.
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply .github/skills/android-reviewer/SKILL.md:22]: This is a pre-existing instruction that this PR did not introduce. Our change actually improves the situation: "if a suggestion cannot be posted inline, omit it entirely rather than burying it in the summary" -- the net effect is fewer bogus comments, not more.

**Review by simonrozsival — APPROVED** [review]
